builtin/logical/database/backend.go

```diff
@@ -57,6 +60,20 @@ func Factory(ctx context.Context, conf *logical.BackendConfig) (logical.Backend,
 	b.credRotationQueue = queue.New()
 	// Load queue and kickoff new periodic ticker
 	go b.initQueue(b.queueCtx, conf, conf.System.ReplicationState())
+
+	// collect metrics on number of plugin instances
+	pluginInstanceGaugeProcess, err := metricsutil.NewGaugeCollectionProcess(
+		[]string{"secrets", "database", "backend", "pluginInstances", "count"},
+		[]metricsutil.Label{},
+		b.collectPluginInstanceGaugeValues,
+		metrics.Default(),
+		configutil.UsageGaugeDefaultPeriod, // TODO: we seem to have no way to access the actual configured values for these?
+		configutil.MaximumGaugeCardinalityDefault,
+		b.logger)
+	if err != nil {
+		return nil, err
+	}
+	go pluginInstanceGaugeProcess.Run()
```

Comment:
This goroutine is leaky since we never call `Stop()` to return from it. A backend's lifecycle may end without the vault process ending on state changes like active to standby, going from unsealed to sealed, or on unmount. We may want to call `Stop` within `databaseBackend.clean()`.

Reply:
Good catch. I'll add that to `clean()`